specs/codecs/wac/spec.md

```diff
@@ -0,0 +1,192 @@
+---
+title: "WAC Specification"
+navTitle: "Spec"
+---
+
+# Specification: WAC
+
+**Status: Prescriptive - Exploratory **
+
+* [Format](#format)
+* [Implementations](#implementations)
+    * [Go](#go)
+    * [Rust/WASM](#rust/wasm)
+
+WAC supports bidirectional transport to and from the [IPLD Data Model].
+In that way it is both a "complete" IPLD Data Model representation and "fitted" to the IPLD Data Model.
+Terminology taken from [Codecs and Completeness].
+
+It takes some inspiration from [Simple DAG], but differs in ways designed to make it reliably round-trip through the IPLD Data Model without loss of fidelity.
+
+The initial motivating use case for WAC is to be a codec that can reliably transport IPLD Data Model data across WebAssembly boundaries while also not being particularly complex to implement.
+
+## Format
+
+### Spec
+
+This format is a series of typing tokens, constant tokens, and inline value data.
+
+Typing tokens are proceeded with the value data for that type.
+
+Every type value can be parsed knowing only the type and without any outside context
+like the container or positional delimiters.
+
+#### Tokens
+
+TODO: Make constants in code match spec
+
+Note: The existing implementations of WAC in https://github.com/aschmahmann/wasm-ipld (Go + Rust/WASM) have different token numbers that we'll need to synchronize before merge.
+
+The tokens for the codec roughly correspond with the [IPLD Data Model Kinds]
+
+They are:
+	Null   = 0
+	True   = 1
+	False  = 2
+	Int    = 3
+	NInt   = 4
+	Float  = 5
+	String = 6
+	Bytes  = 7
+	List   = 8
+	Map    = 9
+	Link   = 10
+
+| Int | Token |
+|---|---|
+| 0 | TYPE_LINK |
+| 1 | TYPE_INTEGER |
+| 2 | TYPE_NEGATIVE_INTEGER |
+| 3 | TYPE_FLOAT |
+| 4 | TYPE_NEGATIVE_FLOAT |
+| 5 | TYPE_STRING |
+| 6 | TYPE_BINARY |
+| 7 | TYPE_MAP |
+| 8 | TYPE_LIST |
+| 9 | VALUE_NULL |
+| 10 | VALUE_TRUE |
+| 11 | VALUE_FALSE |
+
+#### TYPE_LINK
+
+```
+| 0 | CID |
+```
+
+Note: Simple-DAG put a VARINT_LENGTH before the CID indicating how long it was.
+
+CIDs are self-delimiting so this didn't seem necessary.
```

Comment:
Thoughts?

Reply:
+1 on this choice, it may have been left off Simple DAG because at the time we didn't have simple tools to pluck it out without writing an initial-bytes decoder (we've since added one to both Go and JS impls) so it didn't occur as an obvious thing to do